OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -129,6 +125,11 @@ void AddVisibility(Actor a, ref CPos[] visible)
 			foreach (var c in visible)
 			{
 				var uv = c.ToMPos(map);
+
+				// Force cells outside the visible bounds invisible
+				if (!map.Contains(uv))
+					continue;
```

Comment:
I'll add a similar check to the remove case.  We need to store all of the cells in the visibility cache so that we can generate the correct visibility if the map bounds are changed.